package/src/index.ts

```diff
@@ -70,6 +114,255 @@ export class CandleClient {
+    query: AssetAccountQuery = {}
+  ): Promise<AssetAccount[]> {
+    const accounts = await this.candle.getAssetAccounts(query);
+    return accounts.map(this.convertToAssetAccount);
+  }
+
+  public async getTrades(
+    query: {
+      gainedAssetKind?: TradeQueryAssetKind;
+      lostAssetKind?: TradeQueryAssetKind;
+      counterpartyKind?: "merchant" | "user" | "service";
+    } & TradeQuery = {}
+  ): Promise<
+    {
+      dateTime: string;
+      state: TradeState;
+      counterparty: Counterparty;
+      lost: TradeAsset;
+      gained: TradeAsset;
+    }[]
+  > {
+    const trades = await this.candle.getTrades(query);
+    return trades.map(({ dateTime, counterparty, gained, lost, state }) => ({
+      dateTime,
+      state,
+      counterparty: this.convertToCounterparty(counterparty),
+      lost: this.convertTradeAsset(lost)!,
+      gained: this.convertTradeAsset(gained)!,
+    }));
+  }
+
+  public async getTradeQuotes(request: {
+    linkedAccountIDs?: string;
+    gained:
+      | ({
+          assetKind: "nothing";
+        } & NothingAssetQuoteRequest)
+      | ({ assetKind: "transport" } & TransportAssetQuoteRequest)
+      | ({ assetKind: "fiat" } & FiatAssetQuoteRequest)
+      | ({ assetKind: "stock" | "crypto" } & MarketAssetQuoteRequest);
+  }): Promise<
+    {
+      gained?: TradeAsset;
+      lost?: TradeAsset;
+    }[]
+  > {
+    let gainedRequest: TradeAssetQuoteRequest;
+
+    switch (request.gained.assetKind) {
+      case "fiat":
+        gainedRequest = { fiatAssetQuoteRequest: request.gained };
+        break;
+      case "stock":
+      case "crypto":
+        gainedRequest = { marketAssetQuoteRequest: request.gained };
+        break;
+      case "transport":
+        gainedRequest = { transportAssetQuoteRequest: request.gained };
+        break;
+      case "nothing":
+        gainedRequest = { nothingAssetQuoteRequest: request.gained };
+        break;
+    }
+
+    const quotes = await this.candle.getTradeQuotes({
+      linkedAccountIDs: request.linkedAccountIDs,
+      gained: gainedRequest,
+    });
+
+    return quotes.map((quote) => {
+      return {
+        gained: this.convertTradeAsset(quote.gained),
+        lost: this.convertTradeAsset(quote.lost),
+      };
+    });
+  }
+
+  private convertTradeAsset(
+    tradeAsset: InternalTradeAsset
+  ): TradeAsset | undefined {
+    if (tradeAsset.fiatAsset !== undefined) {
+      return {
+        assetKind: "fiat",
+        amount: tradeAsset.fiatAsset.amount,
+        currencyCode: tradeAsset.fiatAsset.currencyCode,
+        linkedAccountID: tradeAsset.fiatAsset.linkedAccountID,
+        serviceAccountID: tradeAsset.fiatAsset.serviceAccountID,
+        service: tradeAsset.fiatAsset.service,
+        serviceTradeID: tradeAsset.fiatAsset.serviceTradeID,
+      };
+    } else if (tradeAsset.marketTradeAsset !== undefined) {
+      return {
+        assetKind: tradeAsset.marketTradeAsset.assetKind as "stock" | "crypto",
```

Comment:
#req don't use `as`–you're just bypassing the compiler, it doesn't actually type-check like `as` in Swift does. 

What I'd do here is just make this an enum in the Nitro types file–Nitro only errors out on single-member enums, this one should work fine

Reply:
No this errors out too.